mochimochi/classifier/binary/pa.hpp

```diff
@@ -34,23 +35,27 @@ public :
 
     // int select : switching the PA algorithm
     // 0 : PA
-    // 1 : PA-1
-    // 2 : PA-2
+    // 1 : PA-I
+    // 2 : PA-II
     switch(kSelect) {
     case 0 :
-      _compute_tau = [](const auto value, const auto loss) {
-        return loss / std::pow(std::abs(value), 2);
+      _compute_tau = [=](const auto value, const auto loss) {
+        /* Check for divide by zero in which case return zero for tau. */
+        /* If "value" is non-zero then proceed with division and return result. */
+        return (value == 0) ? 0 : loss / std::pow(std::abs(value), 2);
       };
       break;
     case 1 :
       _compute_tau = [=](const auto value, const auto loss) {
+        /* Possible divide by zero situation here if "value" is zero resulting in pa = inf. */
+        /* However, the inf is ignored when doing std::min(kC, pa). */
```

Comment:
How do we know it's always +inf and not -inf?

Reply:
It seems like -inf would only happen if loss is negative and I don't believe that is possible. However, since I can't give you a certain answer then maybe I should explicitly check if value == 0 and return kC in case it is and than just put that min part in the else block.

Reply:
Lets put a TODO here to check later and not block the merge!

Reply:
I already made the change and checked it in 🙈.

Reply:
It's ok, let's merge